ddl/placement/rule.go

```diff
@@ -200,12 +206,29 @@ func NewRules(role PeerRoleType, replicas uint64, cnstr string) ([]*Rule, error)
 		}
 
 		for labels, cnt := range constraints2 {
-			labelConstraints, err := NewConstraints(strings.Split(labels, ","))
+			innerLabels := strings.Split(labels, ",")
+			overrideRole := role
+			newLabels := make([]string, 0, len(innerLabels))
+			for _, str := range innerLabels {
+				if strings.HasPrefix(str, attributePrefix) {
+					switch str[1:] {
+					case attributeEvictLeader:
+						if role == Voter {
+							overrideRole = Follower
+						}
+					default:
+						return rules, fmt.Errorf("%w: unsupported attribute '%s'", ErrUnsupportedConstraint, str)
+					}
+					continue
+				}
+				newLabels = append(newLabels, str)
+			}
+			labelConstraints, err := NewConstraints(newLabels)
```

Comment:
LGTM, but what about adding this into `NewConstraints`?

Reply:
currently, the constraint has no `role` semantics, so I still put it outer the `NewConstraints`.